src/Text/Pandoc/Readers/Org.hs

```diff
@@ -472,7 +483,7 @@ rawBlockContent (indent, blockType) = try $
   unlines . map commaEscaped <$> manyTill indentedLine blockEnder
  where
    indentedLine = try $ ("" <$ blankline) <|> (indentWith indent *> anyLine)
-   blockEnder = try $ indentWith indent *> stringAnyCase ("#+end_" <> blockType)
+   blockEnder = try $ skipSpaces *> stringAnyCase ("#+end_" <> blockType)
```

Comment:
+++ Albert Krewinkel [Apr 26 16 05:01 ]:
>   Maybe later, if [3]@jgm agrees, we could add a reader-option to allow
>   nesting, your current changes would come in handy then.

Instead, I would suggest you try to persuade org-mode
maintainers to change Emacs' behavior.